src/main/java/org/jenkinsci/plugins/saml/BundleKeyStore.java

```diff
@@ -0,0 +1,277 @@
+package org.jenkinsci.plugins.saml;
+
+import sun.security.x509.AlgorithmId;
+import sun.security.x509.CertificateAlgorithmId;
+import sun.security.x509.CertificateSerialNumber;
+import sun.security.x509.CertificateValidity;
+import sun.security.x509.CertificateVersion;
+import sun.security.x509.CertificateX509Key;
+import sun.security.x509.X500Name;
+import sun.security.x509.X509CertImpl;
+import sun.security.x509.X509CertInfo;
```

Comment:
After taking a look to bouncycastle-api-plugin, it does not implement how to create a chain of certificates from a keypair nor create a certificate from a keypair I will have to implement it and the only way to create a certificate object is from an ASN1 input stream so I will have also to implement a weird code with temp files to create the certificates and the ASN1 input, so I think in this case bouncycastle-api-plugin does not add worth and make things more complicated,  bouncycastle-api-plugin is designed to import/export key and certificates from files or to files does not have anything to manage certificates with code (easy way) or keystores.